src/System.Private.CoreLib/shared/Interop/Unix/System.Native/Interop.ReadDir.cs

```diff
@@ -3,100 +3,87 @@
 // See the LICENSE file in the project root for more information.
 
 using System;
+using System.Diagnostics;
 using System.Runtime.InteropServices;
-using System.Threading;
+using System.Text;
 using Microsoft.Win32.SafeHandles;
 
 internal static partial class Interop
 {
     internal static partial class Sys
     {
-        private static readonly int s_readBufferSize = GetReadDirRBufferSize();
+        internal static int ReadBufferSize { get; } = GetReadDirRBufferSize();
```

Comment:
Nit: This cache has dubious value. Just calling `GetReadDirRBufferSize` in the few places where this is needed would be just fine.